### PR TITLE
[FLINK-22136][e2e] Console output changed to exception in DataStreamA…

### PR DESCRIPTION
**Pull request by akalash**
…llroundTestProgram

## What is the purpose of the change

*Changes in e2e combinations*

## Brief change log

  - *Even parallelism was added into  run-nightly-tests.sh*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by akalash** [comment]
@flinkbot run azure

**Comment by akalash** [comment]
@flinkbot run azure

**Comment by akalash** [comment]
@flinkbot run azure

**Comment by akalash** [comment]
@flinkbot run azure

File: flink-end-to-end-tests/flink-datastream-allround-test/src/main/java/org/apache/flink/streaming/tests/DataStreamAllroundTestProgram.java [hunk 1]
```diff
@@ -113,7 +113,7 @@ public static void main(String[] args) throws Exception {
                                                                     .getInnerPayLoad()
                                                                     .getSequenceNumber()
                                                             == (event.getSequenceNumber() - 1)) {
-                                                System.out.println(
+                                                throw new Exception(
```
Comment (AHeise) [inline flink-end-to-end-tests/flink-datastream-allround-test/src/main/java/org/apache/flink/streaming/tests/DataStreamAllroundTestProgram.java:116]: I wonder if giving some information (expected actual) would help debugging. Not too sure.

File: flink-end-to-end-tests/run-nightly-tests.sh [hunk 2]
```diff
@@ -103,15 +103,15 @@ run_test "Resuming Savepoint (rocks, scale down, rocks timers) end-to-end test"
 
 run_test "Resuming Externalized Checkpoint (hashmap, async, no parallelism change) end-to-end test" "$END_TO_END_DIR/test-scripts/test_resume_externalized_checkpoints.sh 2 2 hashmap true true" "skip_check_exceptions"
```
Comment (AHeise) [inline flink-end-to-end-tests/run-nightly-tests.sh:104]: Commit message should be adjusted.

**Review by AHeise — APPROVED** [review]
Thank you very much for the fixes. I left two comments for small improvements.

**Comment by akalash** [comment]
@AHeise, I've made the changes as you suggested.

**Pull request merged**